### PR TITLE
FS-4666 : while application update check is round is closed or not 

### PR DESCRIPTION
**Pull request by nuwan-samarasinghe**
While user updating the application and round closed user responses will be saved & redirect them to the fund close page, but there will be trigger to that behaviour.

https://dluhcdigital.atlassian.net/browse/FS-4666

### CODE REVIEW

File: db/queries/statuses/queries.py [hunk 1]
```diff
@@ -243,3 +245,10 @@ def update_statuses(application_id: str, form_name: str, is_summary_page_submitt
 
     update_application_status(application, round.feedback_survey_config)
     db.session.commit()
+    if isinstance(round.deadline, datetime):
```
Comment (srh-sloan) [inline db/queries/statuses/queries.py:248]: It doesn't feel like this belongs in this function - maybe have a separate one that determines round status.

File: db/queries/statuses/queries.py [hunk 2]
```diff
@@ -243,3 +245,10 @@ def update_statuses(application_id: str, form_name: str, is_summary_page_submitt
...
+        deadline = datetime.strptime(round.deadline.strftime("%Y-%m-%dT%H:%M:%S"), "%Y-%m-%dT%H:%M:%S")
+    else:
```
Comment (srh-sloan) [inline db/queries/statuses/queries.py:250]: Why do we need this `if`statement? We have written the time to the database, what format do we use to write it?

File: db/queries/statuses/queries.py [hunk 3]
```diff
@@ -243,3 +245,10 @@ def update_statuses(application_id: str, form_name: str, is_summary_page_submitt
...
+        deadline = datetime.strptime(round.deadline, "%Y-%m-%dT%H:%M:%S")
+    if datetime.now() > deadline:
+        return False
+    return True
```
Comment (srh-sloan) [inline db/queries/statuses/queries.py:254]: See above - doesn't make sense for a function called `update_round_statuses` to return `Boolean`

File: api/routes/application/routes.py [hunk 4]
```diff
@@ -156,7 +156,10 @@ def put(self):
             "is_summary_page_submit": request_json["metadata"].get("isSummaryPageSubmit", False),
         }
         try:
-            updated_form = update_form(**form_dict)
+            updated_form, round_status = update_form(**form_dict)
+            if not round_status:
```
Comment (srh-sloan) [inline api/routes/application/routes.py:160]: More readable if this was called `is_round_open` or similar

File: tests/test_routes.py [hunk 5]
```diff
@@ -580,6 +583,33 @@ def test_complete_form(flask_test_client, seed_application_records):
...
+            "isSummaryPageSubmit": True,
+        },
+    }
+    response = flask_test_client.put(
+        "/applications/forms",
+        json=section_put,
+        follow_redirects=True,
+    )
+    assert response.status_code == 301
+
```
Comment (srh-sloan) [inline tests/test_routes.py:611]: Nice test 👍

**Review by srh-sloan — CHANGES_REQUESTED** [review]

File: tests/conftest.py [hunk 6]
```diff
@@ -260,9 +261,9 @@ def generate_mock_round(fund_id: str, round_id: str) -> Round:
         id=round_id,
         fund_id=fund_id,
         short_name="TEST",
-        opens=datetime.strptime("2023-01-01 12:00:00", "%Y-%m-%d %H:%M:%S"),
-        deadline=datetime.strptime("2023-01-31 12:00:00", "%Y-%m-%d %H:%M:%S"),
-        assessment_deadline=datetime.strptime("2023-03-31 12:00:00", "%Y-%m-%d %H:%M:%S"),
+        opens=(datetime.now() - timedelta(days=30)).strftime("%Y-%m-%dT%H:%M:%S"),
+        deadline=(datetime.now() + timedelta(days=5)).strftime("%Y-%m-%dT%H:%M:%S"),
+        assessment_deadline=(datetime.now() + timedelta(days=10)).strftime("%Y-%m-%dT%H:%M:%S"),
```
Comment (srh-sloan) [inline tests/conftest.py:266]: ⭐

**Review by srh-sloan — APPROVED** [review]
Nice, good work 👍

File: db/queries/statuses/queries.py (hunk 1 above)
Comment (nuwan-samarasinghe) [reply db/queries/statuses/queries.py]: Fixed

File: db/queries/statuses/queries.py (hunk 2 above)
Comment (nuwan-samarasinghe) [reply db/queries/statuses/queries.py]: Fixed

File: db/queries/statuses/queries.py (hunk 3 above)
Comment (nuwan-samarasinghe) [reply db/queries/statuses/queries.py:261]: Fixed

File: api/routes/application/routes.py (hunk 4 above)
Comment (nuwan-samarasinghe) [reply api/routes/application/routes.py]: Fixed

**Pull request merged**